core/src/main/java/bio/terra/pearl/core/service/survey/AnswerService.java

```diff
@@ -42,4 +48,32 @@ public Optional<Answer> findForEnrolleeByQuestion(UUID enrolleeID, String survey
     public void deleteByResponseId(UUID responseId) {
         dao.deleteByResponseId(responseId);
     }
+
+    @Override
+    @Transactional
+    public Answer create(Answer answer) {
+        validateAnswer(answer);
+        return super.create(answer);
+    }
+
+    @Override
+    @Transactional
+    public Answer update(Answer answer) {
+        validateAnswer(answer);
+        return super.update(answer);
+    }
+
+    private void validateAnswer(Answer answer) {
+        answer.inferTypeIfMissing();
+        if (answer.getFormat() == null) {
+            answer.setFormat(AnswerFormat.NONE);
+        }
+
+        if (answer.getFormat().equals(AnswerFormat.FILE_NAME)) {
+            System.out.println(answer.getEnrolleeId());
+            System.out.println(answer.getStringValue());
+            participantFileService.findByEnrolleeIdAndFileName(answer.getEnrolleeId(), answer.getStringValue())
+                    .orElseThrow(() -> new IllegalArgumentException("File not found for answer"));
+        }
+    }
```

Comment:
Since we have the new "format" column, this method will enforce the format. For now it's just file name, but if we get, e.g., date formats, then this will also ensure that the string actually conforms to that format.